ansible/roles/repo/tasks/redhat.yaml

```
@@ -1,16 +1,53 @@
 ---
+- set_fact:
+   rhsm_user: "{{ lookup('env', 'RHSM_USER' ) | ternary (lookup('env', 'RHSM_USER' ), rhsm_user) }}"
 
-- name: RHEL subscription
+- set_fact:
+   rhsm_password: "{{ lookup('env', 'RHSM_PASS' ) | ternary (lookup('env', 'RHSM_PASS' ), rhsm_password) }}"
+
+- set_fact:
+   rhsm_activation_key: "{{ lookup('env', 'RHSM_ACTIVATION_KEY' ) | ternary (lookup('env', 'RHSM_ACTIVATION_KEY' ), rhsm_activation_key) }}"
+
+- set_fact:
+   rhsm_activation_key: "{{ lookup('env', 'RHSM_ACTIVATION_KEY' ) | ternary (lookup('env', 'RHSM_ACTIVATION_KEY' ), rhsm_activation_key) }}"
+
+- set_fact:
+   rhsm_org_id: "{{ lookup('env', 'RHSM_ORG_ID' ) | ternary (lookup('env', 'RHSM_ORG_ID' ), rhsm_org_id) }}"
+
+- name: RHEL subscription using username and password
   redhat_subscription:
     state: present
-    username: "{{ lookup('env', 'RHSM_USER') }}"
-    password: "{{ lookup('env', 'RHSM_PASS') }}"
+    username: "{{ rhsm_user }}"
+    password: "{{ rhsm_password }}"
     auto_attach: true
     force_register: true
+    release: "{{ ansible_distribution_version }}"
+  register: rheluser
+  when:
+    - ansible_distribution == 'RedHat'
+    - rhsm_user | length > 0
+    - rhsm_password | length > 0
+
+- name: RHEL subscription using org_id and activationkey
+  redhat_subscription:
+    state: present
+    org_id: "{{ rhsm_org_id }}"
+    activationkey: "{{ rhsm_activation_key }}"
+    force_register: true
+    release: "{{ ansible_distribution_version }}"
+  register: rhelorg
   when:
-    - lookup('env', 'RHSM_USER') | length > 0
-    - lookup('env', 'RHSM_PASS') | length > 0
     - ansible_distribution == 'RedHat'
+    - rhsm_org_id | length > 0
+    - rhsm_activation_key | length > 0
+
+# we need rhel-8-base-eus-rpms for kernel headers to support gpus
+- name: RHEL subscription refresh
+  shell: subscription-manager config --rhsm.manage_repos=1 && subscription-manager repos --enable=rhel-8-for-x86_64-baseos-eus-rpms && subscription-manager refresh && subscription-manager attach --auto
+  when:
+    - ansible_distribution == 'RedHat'
+    - rhelorg != None or rheluser != None
+    - ansible_distribution_major_version == '8'
```

Should we only enable this when GPU is enabled? Or move this task `ansible/roles/gpu/tasks/nvidia-gpu-RHEL.yaml`?
```suggestion
  when:
    - ansible_distribution == 'RedHat'
    - rhelorg != None or rheluser != None
    - ansible_distribution_major_version == '8'
    - gpu is defined
    - "'nvidia' in gpu.types"
```